### PR TITLE
2022-04-11-spellcheck_clinical_en_3_0

### PR DESCRIPTION
**Pull request by jsl-models**

### CODE REVIEW

**Comment by murat-gunay** [comment]
Sample texts and all page information (including descriptions) are taken from existing model page.

**Comment by vkocaman** [comment]
@murat-gunay why do we use recursiveTokenizer here ? in addition, lets also create a pretrained pipeline and uplaod for the same

**Comment by murat-gunay** [comment]
@vkocaman
Actually I just took the spellcheck pipeline (including recursive tokenizer) from existing model-page. I think that, the previous trainer put it in the pipeline for some reason. I wanted not to change anything. I can replace it with `Tokenizer()` if you want.

I will build spellchecker pipeline and will upload it to models-hub

**Comment by murat-gunay** [comment]
Python/Scala script is updated, `Tokenizer()` annotator is used.

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md [hunk 1]
```diff
@@ -131,4 +129,4 @@ val assembler = new DocumentAssembler()
 
 ## References
 
-MTSamples, i2b2 clinical notes, and several specific medical corpuses.
```
Comment (josejuanmartinez) [inline docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md]: Just a comment - the plural of corpus is `copora`, not corpuses. But no need to change it.

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+           "No cute distress"]
+
+lp.annotate(example)
+```
+```scala
+val assembler = new DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("document")
+
+val tokenizer = new Tokenizer()\
```
Comment (josejuanmartinez) [inline docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md:70]: We don't need the backslashes here

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      .setOutputCol("token")\
+      .setContextChars(["*", "-", "“", "(", "[", "\n", ".","\"", "”", ",", "?", ")", "]", "!", ";", ":", "'s", "’s"])
+
+ val spellChecker = ContextSpellCheckerModel.
+      pretrained("spellcheck_clinical", "en", "clinical/models").
+      setInputCols("token").
+      setOutputCol("checked")
+
+ val pipeline =  new Pipeline().setStages(Array(assembler, tokenizer, spellChecker))
+ val empty_df = spark.createDataFrame([[""]]).toDF("text")
```
Comment (josejuanmartinez) [inline docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md:81]: Is this the way to create and empty DF in scala? I don't think the use of brackets is correct for Scala arrays

**Review by josejuanmartinez — CHANGES_REQUESTED** [review]

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md (hunk 1 above)
Comment (murat-gunay) [reply docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md]: I updated the "corpora"  term also in description section.

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md (hunk 2 above)
Comment (murat-gunay) [reply docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md]: yes you are right. I fixed it

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md (hunk 3 above)
Comment (murat-gunay) [reply docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md:81]: I took it from existing page. Yes you are right. It should be fixed. I will

Comment (murat-gunay) [reply docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md]: fixed

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+lp.annotate(example)
+```
+```scala
+val assembler = new DocumentAssembler()
+      .setInputCol("text")
+      .setOutputCol("document")
+
+val tokenizer = new Tokenizer()
+      .setInputCols(["document"])
```
Comment (josejuanmartinez) [inline docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md:71]: These arrays are also not correct in scala

**Review by josejuanmartinez — CHANGES_REQUESTED** [review]

File: docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md (hunk 4 above)
Comment (murat-gunay) [reply docs/_posts/murat-gunay/2022-04-11-spellcheck_clinical_en_3_0.md]: Fixed. the last minute change of `Tokenizer()` script caused all of this 😄

**Review by josejuanmartinez — APPROVED** [review]

**Pull request merged**